src/renderers/shared/fiber/ReactFiberScheduler.js

```diff
@@ -1372,11 +1375,83 @@ module.exports = function<T, P, I, TI, PI, C, CX, PL>(
     }
   }
 
-  function scheduleUpdate(fiber: Fiber, expirationTime: ExpirationTime) {
-    return scheduleUpdateImpl(fiber, expirationTime, false);
+  function scheduleUpdate(
+    fiber: Fiber,
+    partialState: mixed,
+    callback: (() => mixed) | null,
+    isReplace: boolean,
+    isForced: boolean,
+  ) {
+    let expirationTime;
+    if (expirationContext !== NoWork) {
+      // An explicit expiration context was set;
+      expirationTime = expirationContext;
+    } else if (isPerformingWork) {
+      if (isCommitting) {
+        // Updates that occur during the commit phase should have task priority
+        // by default.
+        expirationTime = Task;
+      } else {
+        // Updates during the render phase should expire at the same time as
+        // the work that is being rendered.
+        expirationTime = nextRenderExpirationTime;
+      }
+    } else {
+      // No explicit expiration context was set, and we're not currently
+      // performing work. Calculate a new expiration time.
+
+      let forceAsync = false;
+      if (fiber.tag === HostRoot) {
```

Comment:
This logic is very top level specific. We should invert this so that it lives in the entry point. Think that we're probably going to have many different things that schedule updates as we iterate on new APIs and in different language runtimes.